Sources/StreamChat/Utils/InternetConnection/InternetConnection.swift

```diff
@@ -6,7 +6,11 @@ import Foundation
 import Network
 
 extension Notification.Name {
-    static let internetConnectionStatusDidChange = Notification.Name("io.getstream.StreamChat.internetConnectionStatus")
+    /// Posted when any the Internet connection update is detected (including quality updates).
+    static let internetConnectionStatusDidChange = Self("io.getstream.StreamChat.internetConnectionStatus")
+    
+    /// Posted only when the Internet connection availability is changed (excluding quality updates).
+    static let internetConnectionAvailabilityDidChange = Self("io.getstream.StreamChat.internetConnectionAvailability")
```

Comment:
In connection recovery handler we are not interested in connection quality updates (`.available(.great) -> .available(.constrained)`) but only in availability changes (`.available` -> `.unavailable`)

There was also an option to cache the previous Internet connection state in the handler but extending the `InternetConnection` API in a way it fits such needs as well felt reasonable 🤷

I would move away from the notification center approach and go with the cancellable subscriptions model but that's the refactoring the team needs to agree upon first